HISTORY.md

```diff
@@ -17,6 +17,10 @@
   * `rocksdb_level_metadata_t` and its and its get functions & destroy function.
   * `rocksdb_file_metadata_t` and its and get functions & destroy functions.
 * Add suggest_compact_range() and suggest_compact_range_cf() to C API.
+* Minor changes to C API for custom comparators.
+
+### Behavior changes
+* DB::Open will now fail if a `Comparator` satisfying `CanKeysWithDifferentByteContentsBeEqual()` is used with a feature that hashes keys or prefixes, such as Bloom filters or hash indices. RocksDB can return wrong results if these hashing-based features are used with a comparator that can treat different byte strings as equal. Because of this change, some existing custom comparators might need to add an override for `CanKeysWithDifferentByteContentsBeEqual()` to return false if appropriate.
```

Comment:
So does "wrong" results mean "RocksDB is supposed to use the customized equal in those hashing-based features" since hash & equal are supposed to matched?

> some existing custom comparators might need to add an override for `CanKeysWithDifferentByteContentsBeEqual()` to return false if appropriate.

Reply:
Hash is not customizable, so if equals is not strictly byte wise then the violated hashing assumption can make Get return Not found falsely, or an old value, or cause other bad behaviors.